File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    const clothesCnt = {};
    for(const [_, type] of clothes) {
        if(type in clothesCnt) clothesCnt[type] += 1;
        else clothesCnt[type] = 2;
    }
    return Object.values(clothesCnt).reduce((acc, cur) => acc * cur, 1) - 1;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> clothesCnt = new HashMap<>();

        for (String[] cloth : clothes) {
            String type = cloth[1];
            clothesCnt.put(type, clothesCnt.getOrDefault(type, 0) + 1);
        }

        int totalCases = 1;
        for (Integer cnt : clothesCnt.values()) {
            totalCases *= (cnt + 1);
        }

        return totalCases - 1;
    }
}
